### ISSUE TITLE
Does generatePrivateChannelToken need a way to set JWT 'sub' claim?

### ISSUE THREAD

**Issue by SteveTherrien** [centrifugal/phpcent#53]
I'm upgrading my application to Centrifugo 4, and I'm getting a `Connection Closed: 3500 invalid token` error when I try to create a new subscription to a channel using a JWT created by `generatePrivateChannelToken()`.

The server logs the following:
```
[INF] token user mismatch channel=$widget:206766-227029_52a5 client=[uuid] tokenUser= user=5555
```

where `user=5555` came from the connection token's **sub** claim.

If I set `"skip_user_check_in_subscription_token": true` in the server config (as described in the [v4 migration steps](https://centrifugal.dev/docs/getting-started/migration_v4#subscription-token-migration)), then it succeeds:
```
[DBG] client subscribed to channel channel=$widget:206766-227029_52a5 client=[uuid] user=5555
```

This migration config option will eventually go away, and I don't want to rely on it.

----

I noticed [generatePrivateChannelToken()](https://github.com/centrifugal/phpcent/blob/929d29444078cc92acf49494f5df3a82f02f893e/src/Client.php#L337) lacks a way of setting a user ID into the JWT's **sub** claim (like is done in the `generateConnectionToken()` function right above it). If I modify it and inject my user's ID directly into the **sub** claim:
```php
$payload['sub'] = '5555';
```

Then everything works without having to set the `skip_user_check_in_subscription_token` config option:
```
[DBG] client subscribed to channel channel=$widget:206766-227029_52a5 client=[uuid] user=5555
```

----

Am I missing something, or does `generatePrivateChannelToken` need a way to set the **sub** claim when subscribing to private channels?

**Comment by FZambia** [comment]
Hello, you are right. I forgot that phpcent contains function to generate tokens.

Meanwhile you can use any JWT library to generate a proper token for Centrifugo.

**Comment by SteveTherrien** [comment]
That works, thanks.

**Comment by FZambia** [comment]
Released [v5.0.0](https://github.com/centrifugal/phpcent/releases/tag/v5.0.0) with changes required to work with Centrifugo v4.
